src/main/kotlin/org/rust/lang/doc/lexer/RustDocHighlightingLexer.flex

```diff
@@ -0,0 +1,190 @@
+package org.rust.lang.doc.lexer;
+
+import com.intellij.lexer.*;
+import com.intellij.psi.tree.IElementType;
+
+import static org.rust.lang.doc.psi.RustDocElementTypes.*;
+import static com.intellij.psi.TokenType.*;
+
+%%
+
+%{
+  private int MAIN_STATE = YYINITIAL;
+
+  public _RustDocHighlightingLexer(boolean isBlock) {
+    this((java.io.Reader)null);
+    MAIN_STATE = isBlock ? IN_BLOCK : IN_EOL;
+  }
+
+  // FIXME(jajakobyly): I think it is possible this will break, though I couldn't make it do so :-)
+  private char CODE_FENCE_DELIM = '\0';
```

Comment:
Yes, this is exactly the reason why I see problem here. The same case caused the bug fixed in  	687cb5a